AEPAssurance/Source/AssuranceEventChunker.swift

```diff
@@ -13,28 +13,57 @@
 import AEPServices
 import Foundation
 
+/// Class that brings the capability to chunk the AssuranceEvent if in need to satisfy the socket size limit.
 struct AssuranceEventChunker {
-    let CHUNK_SIZE = 30 * 1024 // 30KB
 
+    /// The maximum size of data that an `AssuranceEvent` payload can hold after chunking
+    ///
+    /// How did we derive to 30KB?
+    ///  The maximum size of an `AssuranceEvent` to get successfully delivered through the socket is 32KB.
+    ///  AssuranceEvent consist of payload (Dictionary), type(String), vendor(String), metadata (Dictionary), timestamp(Long) and EventNumber (Integer)
+    ///  For the AssuranceEvent to completely fit into the maximum allowed socket size, we safely assign
+    ///    30KB for payload
+    ///    2KB for other fields
+    ///
+    /// Accounting for escape string bloat factor:
+    ///   After chunking the payload into multiple consumable data size, the chunk's are then recreated into corresponding `AssuranceEvent`'s.
+    /// During this process the chunked payload string is escaped and put inside the `chunkData` field of the resulting `Assurance Event`. This
+    /// escaping of string further increases the size of the chunked Data. To accommodate for unknown bloating factor, the chunk size for each event is reduced to 15KB.
+    /// Additionally the factor 0.75 corresponds to size correction due to base64 Encoding of data before sending them over Websocket.
+    let CHUNK_SIZE = (Int) ((15 * 1024) * 0.75) // 15KB
+
+    /// Chunks the given `AssuranceEvent` into multiple socket consumable size AssuranceEvents
+    ///
+    /// The payload field in the `AssuranceEvent` structure has the potential to bottleneck the size limit. Hence only the payload is chopped into multiple smaller chunks.
+    /// Once the payload is chunked, then the chunked data is added in the payload of each AssuranceEvent under the key "chunkData".
+    /// And chunked details are added to the metadata field of the Assurance Event. The chunk details comprises of
+    ///   1. chunkId - Unique Id representing all the chunks of a single event.
+    ///   2. chunkTotal - The total number of chunks to define the original event
+    ///   3. chunkSequenceNumber - Integer Value representing the sequence of chunks
```

Comment:
this value represents a particular chunk's sequence number in the greater event, right?
e.g. - if "this" chunk had a `chunkSequenceNumber` of 3 and `chunkTotal` was 5, you could infer that "this" is chunk 3 of 5

Reply:
You explanation is correct.. I have added more details in this description